.github/workflows/ci-build.yaml

```diff
@@ -149,10 +148,85 @@ jobs:
             done
 
+  gpt:
+    name: Check for ChatGPT suggestions
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event.pull_request.head.repo.full_name == github.repository # Only run on PRs from the same repo
+
+    permissions:
+      contents: read
+      pull-requests: write # so we have permission to comment on pull requests
+      id-token: write # so we can securly fetch the openai secret from the secrets store
+
+    steps:
+      # This is managed here: https://github.com/chainguard-dev/secrets/blob/main/wolfi-dev.tf
+      - uses: google-github-actions/auth@a6e2e39c0a0331da29f7fd2c2a20a427e8d3ad1f # v2.1.1
+        id: auth
+        with:
+          workload_identity_provider: "projects/12758742386/locations/global/workloadIdentityPools/github-pool/providers/github-provider"
+          service_account: "wolfi-dev@chainguard-github-secrets.iam.gserviceaccount.com"
+      - uses: google-github-actions/setup-gcloud@5a5f7b85fca43e76e53463acaa9d408a03c98d3a # v2.0.1
+        with:
+          project_id: "chainguard-github-secrets"
+      - uses: 'google-github-actions/get-secretmanager-secrets@ae0d4054c32840e2ced71207a9df55161ae3debc' # v2.0.0
+        id: secrets
+        with:
+          secrets: |-
+            token:chainguard-github-secrets/openai-secret
+
+      - name: Install gptscript
+        continue-on-error: true
+        uses: cpanato/gptscript-installer@f72d33f0b8f9d6e5c71704a553d7183705a57f74 # v0.2.2
+        if: failure() && steps.secrets.outputs.token != ''
+
+      - name: Run gptscript against the build log
+        id: gptscript
+        continue-on-error: true
+        run:  |
+          for package in ${{needs.changes.outputs.packages}}; do
+            # if error log file exists 
+            if [ -f /temp/${package}.error.log ]; then
+              # Use only the last 150 lines of the build log.
+              tail -n 200 /temp/${package}.error.log >> ${package}.log
+              echo "Calling GTPScript..."
+              echo "GPTScript suggestions to solve the error while building the package ${package}:" >> wolfi-packages.log
+              gptscript gptscripts/wolfi-logs.gpt $(pwd)/${package}.log $package >> wolfi-packages.log
+            fi
+          done
+        if: failure() && matrix.arch == 'x86_64' steps.secrets.outputs.token != ''
+        env:
+          OPENAI_API_KEY: steps.secrets.outputs.token
+
+      - name: Check for chatgpt response
+        id: file_check_gpt_log
+        if: failure() && matrix.arch == 'x86_64' steps.secrets.outputs.token != '' 
+        run: |
+          if test -f "wolfi-packages.log"; then
+            echo "exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "exists=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: GPTScript PR comment diff
+        if: failure() && matrix.arch == 'x86_64'&& steps.secrets.outputs.token != '' && steps.file_check_gpt_log.outputs.exists == 'true'
```

Comment:
Here I only post the hints if `arch==x86_64` and the build failed. But if a different architecture failed and x86_64 passed, then we don't call OpenAI to get any hints. I initially did this to avoid calling OpenAI twice assuming we got the same error on both architectures.

Reply:
Perhaps we might want to have some smart logic here, I went with a simple approach in my initial PR.